op-challenger/fault/responder.go

```diff
@@ -0,0 +1,84 @@
+package fault
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/op-service/txmgr"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// faultResponder implements the [Responder] interface to send onchain transactions.
+type faultResponder struct {
+	log log.Logger
+
+	txMgr txmgr.TxManager
+
+	fdgContractTransactor FaultTransactor
+}
+
+// FaultTransactor is the interface for attacking or defending a claim.
+type FaultTransactor interface {
+	Attack(opts *bind.TransactOpts, parentIndex *big.Int, pivot [32]byte) (*types.Transaction, error)
+	Defend(opts *bind.TransactOpts, parentIndex *big.Int, pivot [32]byte) (*types.Transaction, error)
+}
+
+// NewFaultResponder returns a new [faultResponder].
+func NewFaultResponder(logger log.Logger, txManagr txmgr.TxManager, faultTransactor FaultTransactor) *faultResponder {
+	return &faultResponder{
+		log:                   logger,
+		txMgr:                 txManagr,
+		fdgContractTransactor: faultTransactor,
+	}
+}
+
+// buildTx builds the transaction for the [faultResponder].
+func (r *faultResponder) buildTx(ctx context.Context, response Claim) (*types.Transaction, error) {
+	// Parent Claim index in the contract claimData array.
+	bigIndex := big.NewInt(int64(response.ParentContractIndex))
+
+	// Build the transaction data using the response claim.
+	if response.DefendsParent() {
+		tx, err := r.fdgContractTransactor.Defend(&bind.TransactOpts{}, bigIndex, response.ValueBytes())
```

Comment:
Take a look at https://github.com/ethereum-optimism/optimism/blob/7af004263048d6a8f1de832967769fc2ec7b5945/op-proposer/proposer/l2_output_submitter.go#L313-L320

Basically `TransactOpts` requires a signer & that doesn't work with our transaction management. In addition `TransactOpts` defaults to sending the transaction. As such, we'll have to manually pack the transaction data here.